tests/ops/models/test_application_config.py

```diff
@@ -83,3 +210,301 @@ def test_get_api_config(self, db: Session, example_config_record: Dict[str, Any]
         assert (
             ApplicationConfig.get_api_set_config(db) == example_config_record["api_set"]
         )
+
+    def test_set_config_set_config(self, db):
+        ApplicationConfig.set_config_set_config(db, CONFIG)
+        # check a few specific config properties of different types on the database record
+        db_config = ApplicationConfig.get_config_set_config(db)
+        assert (
+            db_config["notifications"]["notification_service_type"]
+            == CONFIG.notifications.notification_service_type
+        )
+        assert (
+            db_config["notifications"]["send_request_completion_notification"]
+            == CONFIG.notifications.send_request_completion_notification
+        )
+        assert (
+            db_config["execution"]["masking_strict"] == CONFIG.execution.masking_strict
+        )
+        assert db_config["redis"]["port"] == CONFIG.redis.port
+        assert db_config["security"]["cors_origins"] == CONFIG.security.cors_origins
+        # and perform a general check on the config objects
+        assert dumps(db_config) == CONFIG.json()
+
+        # change a few config values, set the db record, ensure it's updated
+        notification_service_type = CONFIG.notifications.notification_service_type
+        execution_strict = CONFIG.execution.masking_strict
+        CONFIG.notifications.notification_service_type = (
+            notification_service_type + "_changed"
+        )
+        CONFIG.execution.masking_strict = (
+            not execution_strict
+        )  # since it's a boolean, switch the value
+
+        # these shouldn't be equal before we update the db record
+        assert (
+            db_config["notifications"]["notification_service_type"]
+            != CONFIG.notifications.notification_service_type
+        )
+        assert (
+            db_config["execution"]["masking_strict"] != CONFIG.execution.masking_strict
+        )
+        # now we update the db record and all values should align again
+        ApplicationConfig.set_config_set_config(db, CONFIG)
+        db_config = ApplicationConfig.get_config_set_config(db)
+        assert (
+            db_config["notifications"]["notification_service_type"]
+            == CONFIG.notifications.notification_service_type
+        )
+        assert (
+            db_config["notifications"]["send_request_completion_notification"]
+            == CONFIG.notifications.send_request_completion_notification
+        )
+        assert (
+            db_config["execution"]["masking_strict"] == CONFIG.execution.masking_strict
+        )
+        assert db_config["redis"]["port"] == CONFIG.redis.port
+        assert db_config["security"]["cors_origins"] == CONFIG.security.cors_origins
+        assert dumps(db_config) == CONFIG.json()
```

Comment:
Same as previous question, does `assert dumps(db_config) == CONFIG.json()` cover all the previous asserts?

Reply:
yeah, i think what i was trying to do here was make sure we don't have any deserialization issues? but that's probably not needed, may be dumb.